doc/plugins/plugin-develop-cn.md

```diff
@@ -0,0 +1,155 @@
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+-->
+[English](plugin-develop.md)
+
+# 目录
+- [**检查外部依赖**](#检查外部依赖)
+- [**插件命名与配置**](#插件命名与配置)
+- [**配置描述与校验**](#配置描述与校验)
+- [**确定执行阶段**](#确定执行阶段)
+- [**编写执行逻辑**](#编写执行逻辑)
+- [**编写测试用例**](#编写测试用例)
+
+
+## 检查外部依赖
+
+如果你的插件，涉及到一些外部的依赖和三方库，请首先检查一下依赖项的版权信息证书并将证书信息加入到我们的COPYRIGHT文件中。
+如果插件需要用到共享内存，需要要 bin/apisix 里面进行申明，例如：
+```nginx
+    lua_shared_dict plugin-limit-req     10m;
+    lua_shared_dict plugin-limit-count   10m;
+    lua_shared_dict prometheus-metrics   10m;
+    lua_shared_dict plugin-limit-conn    10m;
+    lua_shared_dict upstream-healthcheck 10m;
+    lua_shared_dict worker-events        10m;
+
+    # for openid-connect plugin
+    lua_shared_dict discovery             1m; # cache for discovery metadata documents
+    lua_shared_dict jwks                  1m; # cache for JWKs
+    lua_shared_dict introspection        10m; # cache for JWT verification results
+```
+如果插件本身的依赖，需要在nginx初始化启动，则可能需要在 lua/apisix.lua 文件的初始化方法 http_init 中添加逻辑，
+并且可能需要在 bin/apisix 文件中，对 nginx 配置文件生成的部分，添加一些你需要的处理。
+
+
+插件本身提供了 init 方法。方便插件在 init_worker 阶段插件被加载后进行一些初始化处理。 
+
+## 插件命名与配置
+
+给插件取一个很棒的名字，确定插件的加载优先级，然后在 conf/config.yaml 文件中添加上你的插件名。
+例如 key-auth 这个插件，需要在代码里指定插件名称（名称是插件的唯一标识，不可重名），
+在 lua/apisix/plugins/key-auth.lua 文件中可以看到
+```lua
+   local plugin_name = "key-auth"
+```
+在 conf/config.yaml 配置文件中，列出了支持的插件（都是以插件名指定的）
+```yaml
+plugins:                          # plugin list
+  - example-plugin
+  - limit-req
+  - limit-count
+  - limit-conn
+  - key-auth
+  - prometheus
+  - node-status
+  - jwt-auth
+  - zipkin
+  - ip-restriction
+  - grpc-transcode
+  - serverless-pre-function
+  - serverless-post-function
+  - openid-connect
+  - proxy-rewrite
+  - redirect
+```
+## 配置描述与校验
+
+定义插件的配置项，以及对应的 json 的 schema 描述符，并完成对 json 的校验，这样方便对配置的数据规格进行验证，以确保数据的完整性以及程序的健壮性。
+同样，我们以 key-auth 插件为例，看看他的配置数据：
+```json
+ "key-auth": {
+       "key": "auth-one"
+  }
+```
+插件的配置数据比较简单，只支持一个命名为 key 的属性，那么我们看下他的schema描述：
+```lua
+   local schema = {
+       type = "object",
+       properties = {
+           key = {type = "string"},
+       }
+   }
+```
+同时，需要实现check_schema(conf)方法，完成规格的校验。
+```lua
+   function _M.check_schema(conf)
+       return core.schema.check(schema, conf)
+   end
+```
+注：项目已经提供了core.schema.check 公共方法，直接使用即可完成json校验。
+
+## 确定执行阶段
+
+根据业务功能，确定你的插件需要在哪个阶段执行。 不了解openresty生命周期的朋友，建议提前了解下。
+key-auth 是一个认证插件，只要在请求进来之后业务响应之前完成认证即可。该插件在 rewrite 、access 阶段执行都可以，
+项目中是用 rewrite 阶段执行认证逻辑，一般IP准入、接口权限是在 access 阶段完成的。
+
+## 编写执行逻辑
+
+在对应的阶段方法里编写功能的逻辑代码
```

Comment:
TODO or only one line?

Reply:
just one line. do you have some idea ?

Reply:
you can add a todo issue for that.

Reply:
Punctuation is missing.

Reply:
added

> Punctuation is missing.

added.

Reply:
> you can add a todo issue for that.

different plugin has different logic. so just one line desc here.